plugins/types.go

```diff
@@ -0,0 +1,84 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+// plugins package stores all the plugin types used by containerd internally.
+//
+// External plugins should copy from these types and avoid importing this
+// package.
+package plugins
+
+import "github.com/containerd/containerd/plugin"
+
+const (
+	// InternalPlugin implements an internal plugin to containerd
+	InternalPlugin plugin.Type = "io.containerd.internal.v1"
+	// RuntimePlugin implements a runtime
+	RuntimePlugin plugin.Type = "io.containerd.runtime.v1"
```

Comment:
Do we still need this?

Reply:
That might be a question for the Windows folks. The only real use of it is in a comparison for the Windows runtime, but seems that should be in the scope of deprecation and removal.

Reply:
See https://github.com/containerd/containerd/blob/main/task.go#L312 I can file a separate issue for this to fully deprecate the plugin type.